tools/tpm2_load.c

```diff
@@ -54,6 +89,184 @@ static tool_rc load(ESYS_CONTEXT *ectx) {
         ctx.parameter_hash_algorithm);
 }
 
+bool is_PEM(const char *filestr) {
+    FILE *f = fopen(filestr, "rb");
+    char str[11];
+
+    if (f) {
+        // str, 1, 10, f to read 10 blocks of 1 byte size
+        // instead of: str, 10, 1, f to read 1 block of 10 bytes
+        size_t size_read = fread(str, 1, 10, f);
+        fclose(f);
+        if (size_read != 10) {
+            LOG_ERR("Not enough characters read from file for PEM detection");
+            return false;
+        }
+        if (!strncmp(str, PEM_BEGIN_CHARS, (long unsigned)10)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+    LOG_ERR("Unable to open pointer as file, Unable to complete PEM detection");
+    return false;
+}
+
+char getChar(int value) {
+    if (value >=0 && value <= 9) {
+        return ((char) value) + '0';
+    } else if (value >= 10 && value <= 15) {
+        return ((char) value) + 'a';
+    } else {
+        LOG_ERR("Unable to create parent using create primary");
+        return 'Z';
+    }
+}
```

Comment:
Use [this](https://github.com/tpm2-software/tpm2-tools/blob/3bf7a12a879251d52b302507da66fb7af33e20ab/lib/tpm2_util.h#L141) instead.